src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -56,45 +55,50 @@ public class PackageSourceProvider : IPackageSourceProvider
             _configurationDefaultSources = LoadConfigurationDefaultSources(configurationDefaultSources);
         }
 
-        private static IEnumerable<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
+        private static IReadOnlyList<PackageSource> LoadConfigurationDefaultSources(IEnumerable<PackageSource> configurationDefaultSources)
         {
 #if !IS_CORECLR
             // Global default NuGet source doesn't make sense on Mono
-            if (RuntimeEnvironmentHelper.IsMono)
+            if (Common.RuntimeEnvironmentHelper.IsMono)
             {
-                return Enumerable.Empty<PackageSource>();
+                return Array.Empty<PackageSource>();
             }
 #endif
             var packageSourceLookup = new Dictionary<string, IndexedPackageSource>(StringComparer.OrdinalIgnoreCase);
             var packageIndex = 0;
 
             foreach (var packageSource in configurationDefaultSources)
             {
-                packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource, packageSource.Name);
+                packageIndex = AddOrUpdateIndexedSource(packageSourceLookup, packageIndex, packageSource);
             }
 
-            return packageSourceLookup.Values
+            List<PackageSource> defaultSources = new(packageSourceLookup.Count);
+            defaultSources.AddRange(packageSourceLookup.Values
                 .OrderBy(source => source.Index)
-                .Select(source => source.PackageSource);
+                .Select(source => source.PackageSource));
+
+            return defaultSources.AsReadOnly();
         }
 
-        private static Dictionary<string, IndexedPackageSource> LoadPackageSourceLookup(bool byName, ISettings settings)
+        private static List<PackageSource> GetPackageSourceFromSettings(ISettings settings)
```

Comment:
Is there a specific reason why we didn't return `IReadOnlyList` from this method, similar to the `LoadConfigurationDefaultSources` method?

Reply:
We need to add sources from `NuGetDefaults.config` to the list (call the method `AddDefaultPackageSources`) after getting this list from the `ISettings` object.